app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -170,19 +184,37 @@ public void handleTaskUpdate(Integer... vals) {
     public void handleTaskCompletion(AppInstallStatus result) {
         if (result == AppInstallStatus.UpdateStaged) {
             uiController.unappliedUpdateAvailableUiState();
+            if (proceedAutomatically) {
+                launchUpdateInstallTask();
+                return;
+            }
         } else if (result == AppInstallStatus.UpToDate) {
             uiController.upToDateUiState();
+            if (proceedAutomatically) {
+                finishWithResult(RefreshToLatestBuildActivity.ALREADY_UP_TO_DATE);
+                return;
+            }
         } else {
             // Gives user generic failure warning; even if update staging
             // failed for a specific reason like xml syntax
             uiController.checkFailedUiState();
+            if (proceedAutomatically) {
+                finishWithResult(RefreshToLatestBuildActivity.UPDATE_ERROR);
+                return;
+            }
         }
 
         unregisterTask();
```

Comment:
Probably good idea to call `unregisterTask()` even when `proceedAutomatically` is set to `true`